enumerables.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    selfitem.my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    selfitem.my_each { |x| return true if yield(x) }
+    false
+  end
+
+  def my_none?
+    selfitem = self
+    selfitem.my_each { |x| return false if yield(x) }
+    true
+  end
+
+  def my_count
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array.length
+  end
+
+  def my_map(proc = nil)
+    array = []
+    selfitem = self
+    if proc == true
+      selfitem.my_each { |x| array << proc.call(x) }
+    else
+      selfitem.my_each { |x| array << yield(x) }
+    end
+    array
+  end
+
+  def my_inject(parameter = 0)
+    sum = parameter
+    selfitem = self
+    selfitem.my_each { |x| sum = yield(sum, x) }
+    sum
+  end
```

Comment:
enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
      
      NoMethodError:
        undefined method `[]' for 5..50:Range

 enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
      
      LocalJumpError:
        no block given (yield)